app/controllers/auth_controller.rb

```diff
@@ -4,11 +4,47 @@ class AuthController < ApplicationController
   skip_before_filter :verify_authenticity_token, :only => [:access_token]
 
   def verify_logged_in
-    if current_user.nil?
-      session[:sso_callback_params] = params
-      session[:sso_application] = application
-      redirect_to auth_login_path
+
+    action = params[:action]
+
+    if action == 'oauth_authorize'
+
+        #
+        # Calling as SSO provider
+        #
+        # Example params
+        # {"response_type"=>"code", "client_id"=>"localhost", "redirect_uri"=>"http://app.lara.docker/users/auth/cc_portal_localhost/callback", "state"=>"8d300e331c03b2255a4d56269b09fa906f1ff5349e943099", "controller"=>"auth", "action"=>"oauth_authorize"}
+        #
+        #
+        # puts "INFO *** Calling as SSO provider"
+
+        if current_user.nil?
+            session[:sso_callback_params] = params
+            session[:sso_application]     = application
+            redirect_to auth_login_path
+        end
+
+    else
+
+        #
+        # Example params
+        # {"controller"=>"auth", "action"=>"user", "format"=>"json"}
+        #
+        # puts "INFO *** Not calling as SSO provider"
+
+        if session[:omniauth_origin]
+            session[:sso_callback_params] = params
+            session[:sso_application]     = application
```

Comment:
@jsimpson-cc why do we need to save the params if the ominauth_origin is set?
As far as I can tell, we only set the omniauth_origin after we are called back by an oauth provider like Schoology or Google.  And the sso_callback_params are only used when the portal is acting as a provider.